01-fundamentos/secao-07-introducao-es6-testes-unitarios/dia-03-primeiros-passos-jest/mixFunctions.js

```diff
@@ -0,0 +1,75 @@
+function sum(a, b) {
+  if (typeof a !== 'number' || typeof b !== 'number') {
+    throw new Error('parameters must be numbers');
+  }
+  return a + b;
+}
+function myRemove(arr, item) {
+  let newArr = [];
+  for (let index = 0; index < arr.length; index += 1) {
+    if (item !== arr[index]) {
+      newArr.push(arr[index]);
+    }
+  }
+  return newArr;
+}
+function myFizzBuzz(num) {
+  if (typeof num !== 'number') return false;
+  if (num % 3 === 0 && num % 5 === 0) return 'fizzbuzz';
+  if (num % 3 === 0) return 'fizz';
+  if (num % 5 === 0) return 'buzz';
+  return num;
+}
+
+const vogais = ['a', 'e', 'i', 'o', 'u'];
+const code = [1, 2, 3, 4, 5];
+
+function encode(text) {
+  const caracters = text.split('');
+  caracters.forEach((letra, i) => {
+    vogais.forEach((vogal, k) => {
+      caracters[i] = (letra === vogal) ? code[k] : caracters[i];
+    });
+  });
+  return caracters.join('');
+}
+
+function decode(text) {
+  const caracters = text.split('');
+  caracters.forEach((caracter, i) => {
+    code.forEach((num, k) => {
+      caracters[i] = (caracter === num.toString()) ? vogais[k] : caracters[i];
+    });
+  });
+  return caracters.join('');
+}
+
+const techList = (arrList, strName) => {
+  if (arrList.length) {
+    arrList.sort();
+    let newArrList = [];
+    for (const item of arrList) {
+      const obj = {};
+      obj.name = strName;
+      obj.tech = item;
+      newArrList.push(obj);
+    }
+    return newArrList;
+  } else {
+    return 'Vazio!';
+  }
```

Comment:
O `else` nesse caso se torna opcional. Pensando na execução do código: como a condicional `if` está verificando se a array recebida é `!== []`, sempre que a array recebida pela função atender à essa condição, a lógica dentro do `if` será executada, e o `return newArrList` encerrará/concluirá o processamento de dados retornando a array de resposta `newArrList`. A execução é interrompida/concluída antes de chegar ao `return 'Vazio!'`.

Caso o critério da condicional `if` não seja atendido, todo o código dentro do escopo do `if` será ignorado, e a execução vai pular direto para o `return 'Vazio!'`

```suggestion
    return newArrList;
  }
  return 'Vazio!';
```